Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -107,14 +107,14 @@ static ReportingClient* reportingClientForContext(ScriptExecutionContext& script
 }
 
 ContentSecurityPolicy::ContentSecurityPolicy(URL&& protectedURL, ScriptExecutionContext& scriptExecutionContext)
-    : m_scriptExecutionContext(&scriptExecutionContext)
+    : m_scriptExecutionContext(scriptExecutionContext)
     , m_reportingClient { reportingClientForContext(scriptExecutionContext) }
     , m_protectedURL { WTFMove(protectedURL) }
 {
     ASSERT(scriptExecutionContext.securityOrigin());
     updateSourceSelf(*scriptExecutionContext.securityOrigin());
     // FIXME: handle the non-document case.
-    if (is<Document>(m_scriptExecutionContext)) {
+    if (is<Document>(m_scriptExecutionContext.get())) {
```

Comment:
Maybe `dynamicDowncast` here, since you downcast to get its page in the next line?

Reply:
Sure.